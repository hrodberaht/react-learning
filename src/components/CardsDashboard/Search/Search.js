import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { filter } from '../../../actions';

export class Search extends Component {
  render() {
    const { handleChange } = this.props;
    return (
      <div className="search-card">
        <label htmlFor="search">
          <p>Search by name:</p>
          <i className="fa fa-search" />
          <input className="form-input" id="search" type="text" onChange={handleChange} />
        </label>
      </div>
    );
  }
}

Search.propTypes = {
  handleChange: PropTypes.func.isRequired,
};

export default connect(null, { handleChange: filter })(Search);
