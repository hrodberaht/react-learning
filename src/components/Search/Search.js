import React, { Component } from "react";
import PropTypes from 'prop-types';

export default class Search extends Component {
  render() {
    const { filter } = this.props;
    return (
      <div>
        <label htmlFor="search">Search by name:</label>
        <input id="search" type={"text"} onChange={filter}/>
      </div>
    );
  }
}

Search.propTypes = {
  filter: PropTypes.func.isRequired
};
