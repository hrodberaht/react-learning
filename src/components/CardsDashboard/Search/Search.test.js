import { Search } from './Search';

describe('<Search />', () => {
  const defaultProps = {
    handleChange: jest.fn(),
  };
  const setup = buildSetup(Search, defaultProps, null);

  it('should render without crash', () => {
    expect(setup().wrapper).toMatchSnapshot();
  });

  it('should call onChange when typing in input', () => {
    setup().wrapper.find('input').simulate('change', { target: { value: '1' } });
    setup().wrapper.find('input').simulate('change', { target: { value: '2' } });

    expect(setup().props.handleChange.mock.calls.length).toBe(2);
  });
});
