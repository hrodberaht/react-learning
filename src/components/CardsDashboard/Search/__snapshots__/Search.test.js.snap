// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should render without crash 1`] = `
<div
  className="search-card"
>
  <label
    htmlFor="search"
  >
    <p>
      Search by name:
    </p>
    <i
      className="fa fa-search"
    />
    <input
      className="form-input"
      id="search"
      onChange={[Function]}
      type="text"
    />
  </label>
</div>
`;
